BikeRental/BikeRental.Application/Services/LeaseService.cs

```diff
@@ -0,0 +1,88 @@
+using BikeRental.Application.Contracts.Dtos;
+using BikeRental.Application.Interfaces;
+using BikeRental.Application.Mappings;
+using BikeRental.Domain.Interfaces;
+
+namespace BikeRental.Application.Services;
+
+/// <summary>
+/// Application-сервис для работы с договорами аренды велосипедов.
+/// Инкапсулирует бизнес-логику и доступ к репозиторию.
+/// На текущем этапе является тонкой обёрткой над ILeaseRepository.
+/// </summary>
+public sealed class LeaseService(
+    ILeaseRepository leaseRepository,
+    IBikeRepository bikeRepository,
+    IRenterRepository renterRepository) : ILeaseService
+{
+    public async Task<IEnumerable<LeaseDto>> GetAll()
+    {
+        return (await leaseRepository.GetAll()).
+            Select(l => l.ToDto());
+    }
+
+    public async Task<LeaseDto?> GetById(int id)
+    {
+        return (await leaseRepository.GetById(id))?.ToDto();
+    }
+
+    public async Task<LeaseDto> Create(LeaseCreateUpdateDto dto)
+    {
+        var bike = await bikeRepository.GetById(dto.BikeId);
+        if (bike == null)
+        {
+            throw new KeyNotFoundException($"Bike with id {dto.BikeId} not found.");
+        }
```

Comment:
Можно упростить. Аналогично упростите где это возможно.
```suggestion
        var bike = await bikeRepository.GetById(dto.BikeId)
            ?? throw new KeyNotFoundException($"Bike with id {dto.BikeId} not found.");
```